pallets/simple-staking/src/lib.rs

```diff
@@ -0,0 +1,302 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+#[cfg(feature = "runtime-benchmarks")]
+mod benchmarking;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use frame_support::{
+		dispatch::DispatchResultWithPostInfo,
+		pallet_prelude::*,
+		inherent::Vec,
+		traits::{Currency, ReservableCurrency, EnsureOrigin, ExistenceRequirement::KeepAlive},
+		PalletId
+	};
+	use frame_system::pallet_prelude::*;
+	use frame_system::Config as SystemConfig;
+	use frame_support::{
+		sp_runtime::{RuntimeDebug, traits::{AccountIdConversion}},
+		weights::DispatchClass,
+	};
+	use core::ops::Div;
+
+
+	/// Configure the pallet by specifying the parameters and types on which it depends.
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		/// Because this pallet emits events, it depends on the runtime's definition of an event.
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+		/// The currency mechanism.
+		type Currency: ReservableCurrency<Self::AccountId>;
+
+		type UpdateOrigin: EnsureOrigin<Self::Origin>;
+
+		type TreasuryId: Get<PalletId>;
+
+		/// Maximum number of authors that we should have. This is used for benchmarking and is not
+		/// enforced.
+		///
+		/// This does not take into account the invulnerables.
+		type MaxAuthors: Get<u32>;
+
+		/// Maximum number of invulnerables.
+		///
+		/// Used only for benchmarking.
+		type MaxInvulnerables: Get<u32>;
+
+		type WeightInfo: WeightInfo;
```

Comment:
```suggestion
		/// Because this pallet emits events, it depends on the runtime's definition of an event.
		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
		/// The currency mechanism.
		type Currency: ReservableCurrency<Self::AccountId>;
		
		/// Origin that can dictate updating parameters of this pallet.
		type UpdateOrigin: EnsureOrigin<Self::Origin>;

		/// Account Identifier using which the internal treasury account is generated. 
		type TreasuryId: Get<PalletId>;

		/// Maximum number of authors that we should have. This is used for benchmarking and is not
		/// enforced.
		///
		/// This does not take into account the invulnerables.
		type MaxAuthors: Get<u32>;

		/// Maximum number of invulnerables.
		///
		/// Used only for benchmarking.
		type MaxInvulnerables: Get<u32>;

		/// The weight information of this pallet. 
		type WeightInfo: WeightInfo;
```